pkg/controller/release/rolloutblock.go

```diff
@@ -0,0 +1,63 @@
+package release
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	shippererrors "github.com/bookingcom/shipper/pkg/errors"
+	rolloutblockUtil "github.com/bookingcom/shipper/pkg/util/rolloutblock"
+	stringUtil "github.com/bookingcom/shipper/pkg/util/string"
+)
+
+func (s *Scheduler) shouldBlockRollout(rel *shipper.Release) (bool, string, error) {
```

Comment:
Not sure if the semantics make sense to me: on the application level rollout block is a responsibility of the controller itself, and in this controller we delegate an equivalent responsibility to scheduler.
I assume, this might be taken as 2 action points:
1. Does it have to be a responsibility of the scheduler? Can it be moved to the controller?
2. Application-level and release-level blocks don't look very much different, do you think it's possible to have a single component responsible for it and embeddable into distinct controllers?